purchase_order_weight_volume/models/purchase_order.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2023 ooops404
+# License AGPL-3 - See https://www.gnu.org/licenses/agpl-3.0.html
+from odoo import api, fields, models
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    line_weight = fields.Float(
+        "Weight", compute="_compute_line_physical_properties", digits="Stock Weight"
+    )
+    line_volume = fields.Float(
+        "Volume", compute="_compute_line_physical_properties", digits="Volume"
+    )
+
+    @api.depends("product_uom_qty", "product_id")
+    def _compute_line_physical_properties(self):
+        for line in self:
+            line.line_weight = line.product_id.weight * line.product_uom_qty
+            line.line_volume = line.product_id.volume * line.product_uom_qty
```

Comment:
This computation will be wrong depending on the UoM of the line. you can see this PR that compute weight on sale order line : 

https://github.com/OCA/sale-workflow/pull/3414/files#diff-053265b671fff7450b572699fd238c0e01828ae14e5b39101fb04af293e700b1R19-R29